datafusion/datasource/src/source.rs

```diff
@@ -35,7 +35,7 @@ use datafusion_physical_expr_common::sort_expr::LexOrdering;
 
 /// Common behaviors in Data Sources for both from Files and Memory.
 /// See `DataSourceExec` for physical plan implementation
-pub trait DataSource: Send + Sync {
+pub trait DataSource: Send + Sync + Debug {
```

Comment:
probably we can comment why `Debug` is required? 🤔

Reply:
Added a note in a5dbaca52-- the real reason is to help standard debugging.